### PR TITLE
Make ``TenantsBase#getTenantAdmin`` to pure async.

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

Make  ``TenantsBase#getTenantAdmin`` to pure async.

### Modifications

- change sync method to async.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [x] `no-need-doc`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java [hunk 1]
```diff
@@ -90,23 +90,19 @@ public void getTenants(@Suspended final AsyncResponse asyncResponse) {
...
+                .thenCompose(__ -> tenantResources().getTenantAsync(tenant))
+                .thenAccept(tenantInfo -> {
+                    if (!tenantInfo.isPresent()) {
+                        throw new RestException(Status.NOT_FOUND, "Tenant does not exist");
+                    } else {
+                        log.info("[{}] Successfully get Tenant.", clientAppId());
+                        asyncResponse.resume(tenantInfo.get());
+                    }
+                }).exceptionally(ex -> {
+                    log.error("[{}] Failed to get Tenant {}", clientAppId(), ex);
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java:103]: redundant '{}' ?

Comment (mattisonchao) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java:103]: Good catch! PTAL :-)

**Review by RobertIndie — APPROVED** [review]

**Comment by mattisonchao** [comment]
@codelipenghui  @eolivelli  @Technoboy- @Jason918 @Shoothzj @nodece 

PTAL, when you have time. :)

**Review by Technoboy- — APPROVED** [review]
LGTM

**Review by Jason918 — APPROVED** [review]

**Pull request closed without merging**